lib/ncr_dispatcher.rb

```diff
@@ -0,0 +1,6 @@
+class NcrDispatcher < LinearDispatcher
```

Comment:
Yes, Chris mentioned this in both of our meetings with him. I think we should have this in our next demo, whether it is in its current form or if the logic is passed into the existing `LinearDispatcher` method in a block. I don't mind quickly doing the latter, PR into the PR style. ;)